### PR TITLE
[Merged by Bors] - Add config overrides documentation.

### LINKED ISSUE #276: Document supported config overrides
Most settings can be overridden today already even though they are not exposed as fields in the CRD.
This functionality however is hidden and not documented.

Please document all the possible files, CLI and Env (if applicable) that can be overridden.

You can take a look at the Druid PR: https://github.com/stackabletech/druid-operator/pull/154
But please feel free to improve if you find anything.

### PR DESCRIPTION
**Pull request by razvan**
## Description

Fixes #276 

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [x] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: docs/modules/ROOT/pages/usage.adoc [hunk 1]
```diff
@@ -67,88 +67,76 @@ and from outside:
...
-|storePath
-|string
-|A local directory where to cache application data. If set, the history server will store application data on disk instead of keeping it in memory. The data written to disk will be re-used in the event of a history server restart.
-|spark.history.store.path
-|===
+== Configuration & Environment Overrides
+
+The cluster definition also supports overriding configuration properties and environment variables, either per role or per role group, where the more specific override (role group) has precedence over the less specific one (role).
+
+IMPORTANT: Overriding certain properties (such as `spark.port.maxRetries`) which are set by operator can lead to faulty installations.
```
Comment (maltesander) [inline docs/modules/ROOT/pages/usage.adoc:74]: This is not a fitting example, i would go for `SPARK_MASTER_PORT` or `spark.master.port`

**Review by maltesander — CHANGES_REQUESTED** [review]
one minor thing

File: docs/modules/ROOT/pages/usage.adoc (hunk 1 above)
Comment (razvan) [reply docs/modules/ROOT/pages/usage.adoc:74]: Why is it not fitting?  The spark master port is addressed also in the environment section.

Comment (maltesander) [reply docs/modules/ROOT/pages/usage.adoc:74]: Because that property wont lead to faulty installations ever.

Comment (maltesander) [reply docs/modules/ROOT/pages/usage.adoc:74]: and yeah i commented before i was don at the master port... :D

**Review by maltesander — APPROVED** [review]
LGTM.

**Comment by razvan** [comment]
bors merge

**Pull request closed without merging**